variables.tf

```diff
@@ -248,3 +248,107 @@ variable "at_cos_bucket_endpoint" {
   description = "An endpoint for the COS bucket for the Activity Tracker archive. Pass either the public or private endpoint (Only required when var.enable_archive and var.activity_tracker_provision are true)"
   default     = null
 }
+
+########################################################################
+# Activity Tracker Event Routing
+#########################################################################
+
+# COS Targets
+variable "cos_targets" {
+  type = map(object({
+    cos_endpoint = object({
```

Comment:
I think you have only double-wrapped this object so that it fits with the "for_each" you are using for resources, but it is a bit complex for the consumer. I don't really see a need to separate out these variables into two different map elements, with cos_endpoint being an object on its own.
Can we think of a better way to do this, easier for consumer, without these complex objects?

Reply:
External wrapping is actually to allow multiple targets of each type (cos, logdna and event streams) and for internal one (cos_endpoint) I think we can flatten out, previously It had 4 variables now it is reduced to one (target_region) so I think we can do that